### ISSUE TITLE
Different t2/t3 Unit Prices

### ISSUE THREAD

**Issue by notrealn** [Anuken/Mindustry-Suggestions#3176]
### Describe what you would like changed, and why.

Currently, all t2 and t3 units cost the same (excluding the different t1 prices, which are negligible most of the time). 

I do not think this makes any sense. In campaign, you encounter mace much earlier than pulsar, and the reason for that is obvious. Pulsar is the stronger unit, and fighting them with just copper walls and duos is very difficult. However, they cost (almost) the same amount of resources to create. Buffing mace to have the same strength as pulsar and (player controlled) horizon would make it way too hard to deal with in the early game, as you fight them on the second sector with just copper duos. 

In (1v1) pvp, mace and atrax are pretty much never used. One of the biggest reasons for this is how counterable they are. A few hails can completely shut down an attack from mace/atrax, but if you compare the price of a unit factory and making maces to a bunch of hails you can see a huge price difference. Hails are also much quicker to setup than a mace army. 

TLDR: mace is the same price as pulsar, but it sucks a lot more, and can't be buffed as it would make early campaign too hard.

### Describe the changes you want to propose. Include possible alternatives.

Each unit should have a different upgrade price. Yes, this will take some effort to implement, but I believe this is an effective solution to unit balancing. If there is a unit that really sucks, just make it a lot cheaper to compensate. Mace and atrax should be made much cheaper to cover up their weakness of low mobility and range, making them more viable for maybe cheap harassment units. 

I think this is easier than the alternative, which is to make all t2s the same strength and changing all campaign sector waves to match it.

**Comment by itcannotbe** [comment]
Sounds like #2021

**Comment by notrealn** [comment]
oops..........
I swear I did spend some time looking for a duplicate

**Comment by itcannotbe** [comment]
It's fine that one staled some time ago

**Comment by itcannotbe** [comment]
for future reference
#2019
#2020
#2021

**Comment by LeoSko** [comment]
Anuken told to create duplicates if the issue is not addressed in order to remind him. Since this was not, that's fine.

Regarding the issue - yes, that would completely make sense to require more specific resources for a specific unit tree you going for. E.g. push more metaglass for navals, more lead for air, graph for ground, titan for nova tree. That would also encourage a specific gameplay style on a map if there is any reason mapmaker want to make that.

**Comment by genNAowl** [comment]
Please tell me this old PR I made is relevant: https://github.com/Anuken/Mindustry/pull/2405

**Comment by OnyxFinger** [comment]
I am for that idea, as long as it doesn't make the resources radically different. It means that if there are ever different kinds of strategies in gamemodes like PvP (or attack if the map is hard), you will have a good level of commitment to it.
However, these prices shouldn't differ too much, or else your strategy would be instantly seen, and too hard to get away from failing it. It also allows for baiting blablabla... It would be very good for strategic variety, which is a big problem right now.

The only problem is that Anuke probably doesn't like this idea. I think that he didn't go for something similar to this *because* the prices are different and harder to remember (another reason to make those prices somewhat similar).

**Comment by notrealn** [comment]
The other option is trying to make all units of the same tier exactly the same power level. This is fine for t5s, but not really good for low tier units as we might end up with the same situation as crawler. 
In the PR that was created for this (which was closed), anook said “at the moment I don't think it's necessary”. This might mean he is open to changing how the unit upgrading system prices work for future updates.

**Comment by Kyllingene** [comment]
> require more specific resources for a specific unit tree you going for

I agree, although I would for for `lead-ground graphite-air metaglass-naval`, since air units are more-or-less stronger than ground units (and, as a nice bonus, it makes more real-world sense).

**Comment by itcannotbe** [comment]
isn't the point of reconstructors to essential simplify unit creation?

As long as the same items are used I think it won't be a problem, maybe just vary the quantity?

**Comment by OnyxFinger** [comment]
> isn't the point of reconstructors to essential simplify unit creation?
> 
> As long as the same items are used I think it won't be a problem, maybe just vary the quantity?

yes, and to have less blocks for more units (compared to the v5 system)
Maybe a path specific material can be used a little, but the prices don't need to be radical, and can stay easy to remember.
